### PR TITLE
aws lambda application module

### PR DESCRIPTION
**Pull request by xpositivityx**
* Update for identity module in non k8s use case
* Lambda function module which deploys a function, creates a log group and configures lambda to write to that log group.

### CODE REVIEW

**Review by coryodaniel — CHANGES_REQUESTED** [review]
Let’s name this massdriver-application-AWS-lambda per [this discussion](https://massdriverworkspace.slack.com/archives/C03HMFZU4UF/p1661831343340289)

we’d like to get the Massdriver centric modules grouped together and have a nice base set of cloud modules that can be used w/o Massdriver under AWS/ GCP/ etc

File: massdriver-application/aws_identity.tf [hunk 1]
```diff
@@ -6,9 +6,9 @@ locals {
   }
 
   aws_service                      = local.aws_service_map[var.service]
-  aws_eks_oidc_short               = local.is_aws && local.is_kubernetes ? replace(var.kubernetes.cluster_artifact.data.infrastructure.oidc_issuer_url, "https://", "") : null
-  aws_eks_oidc_federated_principal = local.is_aws && local.is_kubernetes ? "arn:aws:iam::${data.mdxc_cloud.current.id}:oidc-provider/${local.aws_eks_oidc_short}" : null
-  aws_eks_assume_role_conditional  = local.is_aws && local.is_kubernetes ? "system:serviceaccount:${var.kubernetes.namespace}:${var.name}" : null
+  aws_eks_oidc_short               = local.is_aws && local.is_kubernetes ? replace(var.kubernetes.cluster_artifact.data.infrastructure.oidc_issuer_url, "https://", "") : ""
```
Comment (WillBeebe) [inline massdriver-application/aws_identity.tf:9]: for what it's worth, here's a thread on when I did this. :)
https://github.com/massdriver-cloud/terraform-modules/pull/25#discussion_r956888413

File: massdriver-application/aws_identity.tf [hunk 2]
```diff
@@ -33,7 +33,7 @@ locals {
 }
 EOF
 
-  aws_federated_principal_assume_role = !local.is_aws ? null : <<EOF
+  aws_federated_principal_assume_role = !local.is_aws && !local.is_kubernetes ? null : <<EOF
```
Comment (WillBeebe) [inline massdriver-application/aws_identity.tf:36]: I'm sure you're aware but dupe of another open PR: https://github.com/massdriver-cloud/terraform-modules/pull/36/files

**Review by WillBeebe — APPROVED** [review]
A few comments but nothing blocking. It's looking good 👍

File: massdriver-application/aws_identity.tf (hunk 1 above)
Comment (coryodaniel) [reply massdriver-application/aws_identity.tf:9]: It ended up breaking in another scenario. Let terraform have it’s nullish ways I suppose. It was evaluation the untrue side of the ternary when true. Very bizarre.

File: massdriver-application/aws_identity.tf (hunk 2 above)
Comment (coryodaniel) [reply massdriver-application/aws_identity.tf:36]: Closed it. Let’s roll lambda templates!!!

**Review by coryodaniel — APPROVED** [review]

**Pull request merged**